File: Solarness.Services/Database/Task.cs
using System;
using System.Collections.Generic;

namespace Solarness.Services.Database;

public partial class Task
{
    public int TaskId { get; set; }

    public string TaskName { get; set; } = null!;

    public string? Description { get; set; }

    public DateOnly? StartDate { get; set; }

    public DateOnly? EndDate { get; set; }

    public string? Status { get; set; }

    public int? ProjectId { get; set; }

    public int? MemberId { get; set; }

    public virtual TeamMember? Member { get; set; }

    public virtual Project? Project { get; set; }
}

File: UI/solarness_mobile/lib/models/Task/task.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:solarness_mobile/models/TeamMember/team_memeber.dart';

import '../Project/project.dart';

part 'task.g.dart';

@JsonSerializable()
class Task {
  int? taskId;
  int? projectId;
  String? taskName;
  String? description;
  DateTime? startDate;
  DateTime? endDate;
  String? status;
  int? memberId;
  Project? project;
  TeamMember? member;


  Task(this.taskId, this.projectId, this.taskName, this.description, this.startDate, this.endDate, this.status,);

  factory Task.fromJson(Map<String, dynamic> json) =>
      _$TaskFromJson(json);

  Map<String, dynamic> toJson() => _$TaskToJson(this);
}
